File: packages/openmemory-py/src/openmemory/i18n/stemmers/en.py
import re

STEM_RULES = [
    (re.compile(r"ies$"), "y"),
    (re.compile(r"ing$"), ""),
    (re.compile(r"ers?$"), "er"),
    (re.compile(r"ed$"), ""),
    (re.compile(r"s$"), ""),
]


def stem_en(tok: str) -> str:
    if len(tok) <= 3:
        return tok
    for pat, rep in STEM_RULES:
        if pat.search(tok):
            st = pat.sub(rep, tok)
            if len(st) >= 3:
                return st
    return tok


def should_stem_en(_tok: str) -> bool:
    return True

File: packages/openmemory-js/src/i18n/stemmers/en.ts
const stem_rules: Array<[RegExp, string]> = [
    [/ies$/, "y"],
    [/ing$/, ""],
    [/ers?$/, "er"],
    [/ed$/, ""],
    [/s$/, ""],
];

export const stem_en = (tok: string): string => {
    if (tok.length <= 3) return tok;
    for (const [pat, rep] of stem_rules) {
        if (pat.test(tok)) {
            const st = tok.replace(pat, rep);
            if (st.length >= 3) return st;
        }
    }
    return tok;
};

export const should_stem_en = (_tok: string): boolean => true;
